test/rule-matches/heading-matches.js

```diff
@@ -0,0 +1,55 @@
+describe('heading-matches', function () {
+  'use strict';
+
+  var fixture = document.getElementById('fixture');
+  var rule;
+
+  beforeEach(function () {
+    rule = axe._audit.rules.find(function (rule) {
+      return rule.id === 'empty-heading';
+    });
+  });
+
+  afterEach(function () {
+    fixture.innerHTML = '';
+  });
+
+  it('is a function', function () {
+    assert.isFunction(rule.matches);
+  });
+
+  it('should return false on elements that are not headings', function () {
+    var div = document.createElement('div');
+    assert.isFalse(rule.matches(div));
+  });
+
+  it('should return true on elements with "heading" in the role', function () {
+    var div = document.createElement('div');
+    div.setAttribute('role', 'heading');
+    assert.isTrue(rule.matches(div));
+
+    div.setAttribute('role', 'slider heading');
+    assert.isTrue(rule.matches(div));
+  });
+
+  it('should return true on regular headings without roles', function () {
+    var h1 = document.createElement('h1');
+    var h2 = document.createElement('h2');
+    var h3 = document.createElement('h3');
```

Comment:
Do these in-memory elements get cleaned up automatically by the test runner?

Reply:
No, but they do by the garbage collector.